### DISCUSSION TITLE
Does OJ Parser support ignoring C comment blocks at beginning of JSON files?

### DISCUSSION THREAD

**Discussion by thadguidry** [ohler55/ojg#168] [General]
I see there is support for ignoring C++ comment lines `//`  , but what about C comment blocks `/* ... */` at beginning of JSON files?

**Comment by ohler55** [comment]
I haven't seen the need for that since comments in JSON are out of spec anyway. Can you provide an example of where you have encountered something like that?

**Reply by thadguidry** [reply]
Indeed, there's been a couple I've run across coming from data.gov and elsewhere sometimes.  They show up enough in JSON files sometimes that even Jackson parser has a boolean option to ignore comments within content (and other feature flags to enable/disable for parsing):

https://github.com/FasterXML/jackson-core/blob/5d8e7ede35d9874626e859d753669ba70cca7256/src/main/java/tools/jackson/core/json/JsonReadFeature.java#L14-L28

https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-core/2.17.0/com/fasterxml/jackson/core/JsonParser.Feature.html

**Reply by thadguidry** [reply]
It's a low priority feature, but would be nice to add to OJ eventually when you can.
And btw, what is usually in the comments sometimes is just lines or paragraphs describing a bit of the schema. (stupid - instead of them actually providing real JSON Schema files etc.)
It's typically from earlier JSON files most are pre-2014 when I guess JSON was more wild west with Schema and annotation handling? dunno.

**Reply by ohler55** [reply]
I'll keep it on the list of features to add. I don't think it will be that tough.

**Reply by ohler55** [reply]
Update: try the c-comment branch. It needs further testing of edge cases but so far it seems to be working. Use the SEN parser though as it it a more tolerant JSON parser.

**Reply by thadguidry** [reply]
Yeah seems to work!  Nice job!

**Reply by ohler55** [reply]
Great, a few more tests and I'll release.

**Reply by ohler55** [reply]
Released v1.22.0
